src/Symfony/Component/HttpFoundation/ParameterBag.php

```diff
@@ -188,6 +188,54 @@ public function getBoolean($key, $default = false)
     }
 
     /**
+     * Returns the parameter value converted to a DateTime object.
+     *
+     * @param string             $key      The parameter key
+     * @param string             $format   The expected date format
+     * @param \DateTime|null     $default  The default value if the parameter key does not exist
+     * @param \DateTimeZone|null $timeZone
+     *
+     * @return \DateTime|null
+     */
+    public function getDate($key, $format = 'Y-m-d', $default = null, $timeZone = null)
+    {
+        if (!$this->has($key)) {
+            return $default;
+        }
+
+        $time = $this->get($key);
+
+        // if the user has specified a timezone then pass that
+        // otherwise do not even attempt to put a value but rather let the runtime decide
+        // the default value by itself
+        // this is in order to ensure compatibility with all php versions since
+        // some accept null as a TimeZone parameter and others do not
+        if ($timeZone !== null) {
+            $result = \DateTime::createFromFormat($format, $time, $timeZone);
+        } else {
+            $result = \DateTime::createFromFormat($format, $time);
+        }
+
+        // Failure to parse the date according to the specified format will return null
+        return false === $result ? null : $result;
+    }
+
+    /**
+     * Returns the parameter value converted to a DateTime object while also parsing the time.
+     *
+     * @param string             $key      The parameter key
+     * @param string             $format   The expected date format
+     * @param \DateTime|null     $default  The default value if the parameter key does not exist
+     * @param \DateTimeZone|null $timeZone
+     *
+     * @return \DateTime|null
+     */
+    public function getDateTime($key, $format = 'Y-m-d H:i:s', $default = null, \DateTimeZone $timeZone = null)
```

Comment:
The format argument could very well be mandatory. As per the OP, I asked for feedback on what could be done with the formats and what could be some defaults. The idea behind this PR is avoid repetitive tasks with handling date parameters on GET requests.